### PR TITLE
🔨 Refactor Export

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## 📝 Description

- Provide a clear summary of the changes and the issue that has been addressed.
- 🛠️ Fixes https://github.com/openvinotoolkit/anomalib/issues/1752

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/deploy/export.py [hunk 1]
```diff
@@ -97,271 +97,254 @@ def convert_center_crop(self) -> None:
...
-        ...     task="segmentation",
-        ... )
-
-    """
-    if not try_import("openvino"):
-        logger.exception("Could not find OpenVINO. Please check OpenVINO installation.")
-        raise ModuleNotFoundError
-
-    import openvino as ov
+class ExportMixin:
```
Comment (ashwinvaidya17) [inline src/anomalib/deploy/export.py:100]: I feel this and `InferenceModule` need a separate file each

File: src/anomalib/models/components/base/anomaly_module.py
Comment (samet-akcay) [inline src/anomalib/models/components/base/anomaly_module.py:1]: what was the agreement about the location of `ExportMixin`? I remember there was a discussion about this but don't remember how it is settled.

**Review by samet-akcay — APPROVED** [review]
lgtm thanks

File: src/anomalib/models/components/base/anomaly_module.py
Comment (ashwinvaidya17) [reply src/anomalib/models/components/base/anomaly_module.py:1]: I think @djdameln wanted this under anomalib components

Comment (samet-akcay) [reply src/anomalib/models/components/base/anomaly_module.py:1]: Do you have any opinion on this?

Comment (ashwinvaidya17) [reply src/anomalib/models/components/base/anomaly_module.py:1]: I also prefer putting this under components as it is only used in `AnomalyModule`

**Comment by CarlosNacher** [comment]
Hi @ashwinvaidya17 @samet-akcay 
One question, perhaps is a Pytorch general doubt, when you `model.to_onnx` (or to_torch for example), the model you are exporting does use the best weights or the last ones? And when in Anomalib we use the `export` method of the `Engine` class, what happens? If the last ones, how should we act to export the best weights taking advantage of the Anomalib's methods?

Thank you so much.

**Comment by CarlosNacher** [comment]
@ashwinvaidya17 got it, thank you!

**Pull request merged**